src/hooks/lib/preview/compounder/useCalcMintGlp.ts

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react"
+import { formatUnits } from "viem"
+import { Address, useContractRead } from "wagmi"
+
+import { getGlpPrice } from "@/lib/api/pricer/getGlpPrice"
+import { useActiveChainId } from "@/hooks/useActiveChainId"
+
+import { GlpVault } from "@/constant/abi"
+import {
+  DAI_ARBI,
+  ETH,
+  FRAX_ARBI,
+  glpVault,
+  LINK_ARBI,
+  UNI_ARBI,
+  USDC_ARBI,
+  USDT_ARBI,
+  WBTC_ARBI,
+  WETH_ARBI,
+} from "@/constant/addresses"
+
+export default function useCalcMintGlp({
+  amount,
+  token,
+  enabled,
+}: {
+  amount: string
+  token: Address
+  enabled: boolean
+}) {
+  const chainId = useActiveChainId()
+  const isArbitrumFamily = chainId === 313371 || chainId === 42161
+
+  if (token == ETH) {
+    token = WETH_ARBI
+  }
+
+  const [glpPrice, setGlpPrice] = useState(0)
+
+  useEffect(() => {
+    getGlpPrice().then((price) => setGlpPrice(price))
+  }, [])
+
+  const tokenPriceWei = useContractRead({
+    chainId,
```

Comment:
GLP is only available on Arbitrum, so we should just use the correct `chainId` in this hook and remove any existing checks against the active chain ID. It doesn't really matter what chain the user is connected to, we can do the lookups.

```
// import from wagmi
chainId: arbitrum.chainId
```